

<head> 
<title>CPDL</title>
</head> 
<body>
<!-- $Id: splash.html,v 3.2 2004/11/29 20:51:01 mccorkle Exp mccorkle $ -->

<H1>CPDL</H1>
Conserved Property Difference Locator
<P>
<IMG SRC="fig1.jpg" WIDTH=30% ALIGN=right>
CPDL is designed to flag positions in a multiple sequence alignment where
two homologous groups of proteins differ either in an amino acid
residue or in a property such as charge or hydrophobicity. The idea is
that such positions are likely to be biologically important for
defining the specific functions of the two
groups.
<P>

The program is described in detail in this open access publication:<BR>
<A HREF="http://www.biomedcentral.com/1471-2105/6/284">BMC Bioinformatics 2005, 6:284 (30 Nov 2005)</A>

<P>
<A HREF="cpdl_form.php"><B>Run CPDL</B></A>
<BR>
<BR>
Here are some examples you might find helpful:<BR>
<TABLE CELLSPACING=5 VALIGN=center>
<TR VALIGN="center">
<TH>Alignment file</TH>
<TH>Top group</TH>
<TH >Output</TH></TH>
</TR>
<TR>
<TD VALIGN="center"><A HREF="Examples/MurE_D.aln">MurE/D</A></TD>
  <TD VALIGN="center">1 to 25</TD> 
  <TD VALIGN="center"><A HREF="Examples/MurE_D.pdf">PDF</A></TD>
</TR>

<TR>
<TD VALIGN="center"><A HREF="Examples/cyclases.msf">ATP cyclases</A></TD>
  <TD VALIGN="center">1 to 20</TD> 
  <TD VALIGN="center"><A HREF="Examples/ATP_GTP_cyc.pdf">PDF</A></TD>
</TR>

<TR>
<TD VALIGN="center"><A HREF="Examples/fad_oh.msf">FAD OH</A></TD>
  <TD VALIGN="center">1 to 2</TD> 
  <TD VALIGN="center"><A HREF="Examples/fad_oh.pdf">PDF</A></TD>
</TR>

</TABLE>
Open the alignment file, save to your machine
<B>as a text file</B>
 and use as input 
<A HREF="cpdl_form.php">here</A>.<BR>
(<I>Note: property track displays were set to "if in main track" for these examples</I>)

<BR>
<BR>

<P>
<B>CPDL4 is OPEN SOURCE!</B><BR>
CPDL4 is a perl rewrite of the original CPDL (implemented in Scheme).
The source code is now available under an 
<A HREF="CPDL4/LICENSE.txt">open source license</A> 
and can be found at <A HREF="https://github.com/sean-mccorkle/CPDL">https://github.com/sean-mccorkle/CPDL</A>.


<BR><BR><BR><BR><BR><BR><BR><BR><BR>

<h3>Explanations</h3>
<BLOCKQUOTE>

<dl>
<dt><A name"afile">Alignment file</A></dt>
<dd> 
CPDL assumes that the input alignment contains sequences from 
two groups of homologous proteins. The input alignment must be organized
 such that the sequences
of the first group occupy rows 1 through N and all members of 
the second group fall below row N.
<BR>
<BR>
CPDL recognizes alignment files generated with programs such as
ClustalW 
(<A HREF="http://www.ebi.ac.uk/clustalw/">http://www.ebi.ac.uk/clustalw/</A>) 
that have been saved as .aln
or .msf only at the present time.  Note that VectorNTI alignment files
can be exported in .msf format and then used in CPDL. Please send an
email to the Webmaster to inquire about additional file types.
<BR>
<BR>
Files can be converted to the proper
format using the ReadSeq tool 
(<A HREF="http://thr.cit.nih.gov/molbio/readseq/">http://thr.cit.nih.gov/molbio/readseq/</A>)
<BR>
<BR>
</dd>

<dt><a name="topg">Top group</a></dt>
<dd>
The input here should be the number N that defines the last sequence in
the first group of your alignment file.
<BR>
<BR>
</dd>

<dt><a name="ofile">Output file name</a></dt>
<dd>
Please name your file it will automatically be tagged as .ps or .pdf
by the program when you choose which to generate.
<BR>
<BR>
</dd>

<dt><a name="track">Track</a></dt> 
<dd>
The tracks are (1) the main sequence track which is
always turned on, followed in descending order down the page: (2)
size, (3) hydrophobicity, (4) charge, (5) polarity, and (6)
aromaticity.
<BR>
<BR>
</dd>

<BR>
<IMG SRC="symb.jpg">
<BR>
<BR>
<dt><a name="size">Size</a></dt>
<dd>
Defined as tiny (G, S, A, CH), small (Cs-s, N, D, T, V) or large (I,
L, M, F, Y, W, H, K, R, E, Q).
</dd>
<BR>


<dt><A name="hydro">Hydrophobicity</a></dt>
<dd>
Defined as hydrophobic (F, Y, W, H, K, T, C, G, A, V, I, L, M) or hydrophilic (P, S, N, D, E, Q, R).
</dd>
<BR>

<dt><A name="hydro">Charge</a></dt>
<dd>
Charged residues are defined as (R, K, H, D, E).
</dd>
<BR>

<dt><A name="polarity">Polarity</a></dt>
<dd>
Polar residues are defined as (Y, W, H, K, R, D, E, Q, N, T, S, Cs-s).
</dd>
<BR>

<dt><A name="aromatic">Aromaticity</a></dt>
<dd>
Aromatic residues are defined as (F, W, Y, H).
</dd>
<BR>

<dt><A name="display">Display Status</a></dt>
<dd>
Defines which property tracks will be included in the output. The
default setting is <I>off</I> for all tracks. Each track can be turned
completely on, in which case the properties for every single residue
in the protein sequence will be displayed. Alternative options allow
the user to display the properties at only those positions where there
are differences in (1) a specific property track, (2) the main
sequence track, or (3) at a red hourglass (<I>picture</I>) position in the main
sequence track.
</dd>
<BR>

<dt><A name="conservation">Conservation level</a></dt>
<dd>
When set to <I>all</I>, requires every member of a class to contain the
same residue at a given position.<BR>
When set to <I>all-but-one</I>, allows one member of a class
to have an alternative residue.
</dd>
<BR>

<dt><A name="gray">Grayscale Function</a></dt> 
<dd>
Defines the level of shading of the residues/properties in the
tracks. The default setting is <I>flat</I> which makes all conserved
residues black and all non-conserved residues medium gray. The
<I>linear</I> setting colors the non-conserved residues along a linear
gradient relative to their frequency in the alignment, while the
<I>exponential</I> setting uses an exponential gradient.
<BR>
<BR>
</dd>

<dt><A name="darkness">Darkness adjustment</a></dt>
<dd>
A number between 0.0 and 1.0, defaulting to 0.25.<BR>
If some features are too faint on your printer output,
try increasing this number.
<BR>
<BR>
</dd>

<dt><A name="ftype">Output Files</a></dt>
<dd>
CPDL generates either Postscript or PDF files. You can download a
program for reading .ps files here (http://www.cs.wisc.edu/~ghost/)
and one for .pdf files here
(http://www.adobe.com/products/acrobat/readstep2.html).
<BR>
<BR>
</dd>

<dt><A name="flags">Flags</a></dt>
<dd>

Closed triangles mark positions at which one group has a conserved
residue that is different from all the residues found at the same
position in the other group. The triangle will be open if any residue
at the same position in the other group matches the conserved residue
in the first group.
<P>

Orange circles mark positions at which one group has at least one
conserved property that is different from the properties found at the
same position in the other group.
<BR>
</dd>

<dl>
</body>
</BLOCKQUOTE>
